yb-voyager/src/transform/transform.go

```diff
@@ -0,0 +1,153 @@
+		transformation can be applied to change those datatypes to supported ones
+	- Rename Table Partitions: In Oracle(ora2pg) case, we might need to rename table partitions		[Future]
+		to match the naming convention of the target database or retain original names.
+
+*/
+
+// In future we can convert this to interface and have separate implementations for each transformation.
+// But for now, kept it simple as a struct with methods
+type Transformer struct {
+}
+
+func NewTransformer() *Transformer {
+	return &Transformer{}
+}
+
+var constraintTypesToMerge = []pg_query.ConstrType{
+	pg_query.ConstrType_CONSTR_PRIMARY, // PRIMARY KEY
+	pg_query.ConstrType_CONSTR_UNIQUE,  // UNIQUE
+	pg_query.ConstrType_CONSTR_CHECK,   // CHECK
+}
+
+/*
+MergeConstraints scans through `stmts` and attempts to merge any
+"ALTER TABLE ... ADD CONSTRAINT" into the corresponding "CREATE TABLE" node.
+
+Error handling: If there is an error, it will be logged and the function will
+return the original `stmts`.
+
+Note: Need to keep the relative ordering of statements(tables) intact.
+Because there can be cases like Foreign Key constraints that depend on the order of tables.
+*/
+func (t *Transformer) MergeConstraints(stmts []*pg_query.RawStmt) ([]*pg_query.RawStmt, error) {
+	// TODO: Ensure removing all the ALTER stmts which are merged into CREATE. No duplicates.
+
+	createStmtMap := make(map[string]*pg_query.RawStmt)
+	for _, stmt := range stmts {
+		stmtType := queryparser.GetStatementType(stmt.Stmt.ProtoReflect())
+		if stmtType == queryparser.PG_QUERY_CREATE_STMT {
+			objectName := queryparser.GetObjectNameFromRangeVar(stmt.Stmt.GetCreateStmt().Relation)
+			createStmtMap[objectName] = stmt
+		}
+	}
+
+	/*
+		Logic to merge constraints into CREATE TABLE
+		For each table, take CREATE TABLE stmt and merge its ALTER TABLE ADD CONSTRAINT stmts
+		(constraint expected: PRIMARY KEY, UNIQUE, FOREIGN KEY, CHECK, DEFAULT)  (Q: what else constraint type can be there?)
+
+		Assumptions
+		1. There is CREATE TABLE stmt for each ALTER TABLE stmt
+		2. There will be only one ALTER TABLE cmd per ALTER TABLE stmt
+
+		Note: Iterating over the original array in such a way that we can keep the relative ordering of statements intact.
+	*/
+	var result []*pg_query.RawStmt
+	for _, stmt := range stmts {
+		stmtType := queryparser.GetStatementType(stmt.Stmt.ProtoReflect())
+		switch stmtType {
+		case queryparser.PG_QUERY_CREATE_STMT:
+			result = append(result, stmt)
+		case queryparser.PG_QUERY_ALTER_TABLE_STMT:
+			objectName := queryparser.GetObjectNameFromRangeVar(stmt.Stmt.GetAlterTableStmt().Relation)
+			alterTableNode := stmt.Stmt.GetAlterTableStmt()
+
+			/*
+				There can be multiple sub commands in an ALTER TABLE stmt
+				Example: ALTER TABLE my_table
+							ADD COLUMN new_col INTEGER,
+							ADD CONSTRAINT my_pkey PRIMARY KEY (id);
+			*/
+			if len(alterTableNode.Cmds) == 0 {
```

Comment:
is this even possible?

Reply:
In case of incomplete or wrong syntax SQLs